src/events/http2.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * @fileoverview Handlers for http2 libraries instrumentation
+ */
+
+const shimmer = require('shimmer');
+const urlLib = require('url');
+const utils = require('../utils.js');
+const tracer = require('../tracer.js');
+const eventInterface = require('../event.js');
+const { isBlacklistURL, isBlacklistHeader } = require('../helpers/events');
+const {
+    isURLIgnoredByUser,
+    resolveHttpPromise,
+    USER_AGENTS_BLACKLIST,
+    URL_BLACKLIST,
+    generateEpsagonTraceId,
+    updateAPIGateway,
+    setJsonPayload,
+} = require('../helpers/http');
+const tryRequire = require('../try_require');
+
+const http2 = tryRequire('http2');
+
+
+/**
+ * http2 module adds in the request and response headers also extra fields with a ':' prefix.
+ * For example :path, :method, etc. When we want to record just the headers, we want to clear
+ * out these fields - using this function.
+ * @param {object} headers data
+ * @returns {object} only real headers without ':'
+ */
+function extractHeaders(headers) {
+    return Object.entries(headers) // Iterate over key-value pairs
+        .filter(header => !header[0].startsWith(':')) // Filter out keys that start with ':'
+        .reduce((obj, header) => { // Rebuild key-value into object using reduce
+            const [key, value] = header;
+            obj[key] = value; // eslint-disable-line no-param-reassign
+            return obj;
+        }, {});
+}
+
+/**
+ * Wraps the http2's module request function with tracing
+ * @param {Function} wrappedFunction The http2's request module
+ * @param {string} authority hostname
+ * @returns {Function} The wrapped function
+ */
+function httpWrapper(wrappedFunction, authority) {
+    return function internalHttpWrapper(headers, options) {
+        let clientRequest = null;
+        let httpEvent = null;
+        let startTime = null;
+        try {
+            const { hostname } = urlLib.parse(authority);
```

Comment:
this should be below the 2 ifs

Reply:
It still can't be - this is being used in the first if